### PR TITLE
- Introducing GeoWebCacheEnvironment

### PR DESCRIPTION
**Pull request by afabiani**
This is releated to the work done on GeoServer

PR: (GEOS-7614) Allowing GeoServer to store property placeholders as templates into the catalog

For more details please read GeoServer GSIP 145

https://github.com/geoserver/geoserver/wiki/GSIP-145

### CODE REVIEW

File: geowebcache/diskquota/jdbc/src/main/java/org/geowebcache/diskquota/jdbc/JDBCConfiguration.java [hunk 1]
```diff
@@ -369,4 +371,34 @@ public String toString() {
...
+            conf.setJNDISource((String) gwcEnvironment.resolveValue(getJNDISource()));
+            ConnectionPoolConfiguration connectionPoolConfig = getConnectionPool();
+            if (connectionPoolConfig != null) {
+                ConnectionPoolConfiguration expConnectionPoolConfig = (ConnectionPoolConfiguration) SerializationUtils.clone(connectionPoolConfig);
+                expConnectionPoolConfig.setDriver((String) gwcEnvironment.resolveValue(connectionPoolConfig.getDriver()));
+                expConnectionPoolConfig.setUrl((String) gwcEnvironment.resolveValue(connectionPoolConfig.getUrl()));
+                expConnectionPoolConfig.setUsername((String) gwcEnvironment.resolveValue(connectionPoolConfig.getUsername()));
+                expConnectionPoolConfig.setPassword((String) gwcEnvironment.resolveValue(connectionPoolConfig.getPassword()));
+                expConnectionPoolConfig.setValidationQuery((String) gwcEnvironment.resolveValue(connectionPoolConfig.getValidationQuery()));
+
```
Comment (afabiani) [inline geowebcache/diskquota/jdbc/src/main/java/org/geowebcache/diskquota/jdbc/JDBCConfiguration.java]: NOTICE that - JDBCConfiguration does not have PORT as configuration parameter. Everything is handled through the JDBC URL.

Comment (aaime) [inline geowebcache/diskquota/jdbc/src/main/java/org/geowebcache/diskquota/jdbc/JDBCConfiguration.java]: Right, good point

**Comment by aaime** [comment]
Looks good to me. @smithkm any opinion?

**Comment by aaime** [comment]
I see no further feedback. The original pull request has been around 25+ days ( https://github.com/GeoWebCache/geowebcache/pull/399 ), I assume there is no time or interest for further review, merging

**Pull request merged**